dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyGraphBuilder.java

```diff
@@ -192,11 +194,16 @@ private DependencyGraphResult buildDependencyGraph(
     } catch (DependencyResolutionException ex) {
       DependencyResult result = ex.getResult();
       node = result.getRoot();
+
+      Set<Artifact> checkedArtifacts = new HashSet<>();
       for (ArtifactResult artifactResult : result.getArtifactResults()) {
         Artifact resolvedArtifact = artifactResult.getArtifact();
+
         if (resolvedArtifact == null) {
           Artifact requestedArtifact = artifactResult.getRequest().getArtifact();
-          artifactProblems.add(createUnresolvableArtifactProblem(node, requestedArtifact));
+          if (checkedArtifacts.add(requestedArtifact)) {
+            artifactProblems.add(createUnresolvableArtifactProblem(node, requestedArtifact));
```

Comment:
Good idea. It might make sense to create our own type for ArtifactProblems rather than an ImmutableSet. Then the checking as to whether we've already created one hides inside that class.

Reply:
The problem was `createUnresolvableArtifactProblem` method needs to traverse the (potentially large) graph. So we need to check whether an artifact is already checked or not before calling the method.